src/dmd/statement.d

```diff
@@ -1680,7 +1680,7 @@ extern (C++) final class SwitchStatement : Statement
         {
             for (auto v = vd; v && v != lastVar; v = v.lastVar)
             {
-                if (v.isDataseg() || (v.storage_class & (STC.manifest | STC.temp)) || v._init.isVoidInitializer())
+                if (v.isDataseg() || (v.storage_class & (STC.manifest | STC.temp) && vd.ident != Id.withSym) || v._init.isVoidInitializer())
```

Comment:
This is required because the `VarDeclaration`s for `with(...)` are now properly marked as temporarys.
(the special treatment for `with` seems odd but that's out of scope for this PR)